sled-agent/src/storage/dump_setup.rs

```diff
@@ -246,6 +292,26 @@ fn below_thresh(
     Ok((below, used))
 }
 
+// zone crate's 'deprecated' functions collide if you try to enable
+// its 'sync' and 'async' features simultaneously :(
```

Comment:
there are other parts of the crate that use the async versions of these functions, and due to the way the `#[cfg()]`'s are used in the snippet you linked, the `*_blocking` versions are inaccessible unless async *isn't* chosen. it'd suffice to either remove the deprecated variants, or if it makes any sense, just make the explicitly-named blocking versions be available unconditionally. (my code here was written sync primarily because the command wrappers in illumos_utils are sync)

Reply:
I pushed a new version of the zone crate that should allow access to both sync and async methods.

(I *also* pushed functions that allow access to the `std::process::Command`, and let the caller decide how it should be run, which I'm gonna use in https://github.com/oxidecomputer/omicron/pull/3442 )